packages/@ourworldindata/grapher/src/timeline/TimelineController.ts

```diff
@@ -65,16 +65,85 @@ export class TimelineController {
         return R.last(this.timesAsc)!
     }
 
-    calculateProgress(time: Time): number {
-        return (time - this.minTime) / (this.maxTime - this.minTime)
+    get timespan(): number {
+        return this.maxTime - this.minTime
+    }
+
+    /**
+     * Determines whether to use equal or proportional spacing for the timeline slider.
+     *
+     * When using equal spacing, each time point is given equal visual space on the slider.
+     * When using proportional spacing, time points are spaced based on their actual values.
+     */
+    @computed get shouldUseEqualSpacing(): boolean {
+        // Few time points always use continuous scale
+        if (this.timesAsc.length < 20) return false
+
+        // Short timespans always use continuous scale
+        if (this.timespan <= 500) return false
+
+        // Calculate gaps between consecutive time points
+        const gaps = R.zipWith(
+            this.timesAsc.slice(1),
+            this.timesAsc,
+            (curr, prev) => curr - prev
+        )
+
+        const sortedGaps = R.sortBy(gaps, (gap) => -gap)
+
+        // Use equal spacing if gaps are highly concentrated (top decile > 50%)
+        const topDecileCount = Math.max(1, Math.floor(gaps.length * 0.1))
+        const topDecileSum = R.sumBy(
+            R.take(sortedGaps, topDecileCount),
+            R.identity()
+        )
```

Comment:
Using `takeFirstBy` can do the same without the full overhead of sorting the array!

```suggestion
        const topDecileSum = R.pipe(
            gaps,
            R.takeFirstBy(gaps.length * .1, [R.identity(), "desc"]),
            R.sum()
        )
```